Mathlib/Data/Subtype.lean

```diff
@@ -3,175 +3,222 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl
 -/
-
-import Std.Tactic.Ext
-import Mathlib.Init.Function
 import Mathlib.Logic.Function.Basic
-import Mathlib.Tactic.Basic
-import Mathlib.Tactic.Coe
+import Mathlib.Tactic.Simps.Basic
+
+/-!
+# Subtypes
+
+This file provides basic API for subtypes, which are defined in core.
+
+A subtype is a type made from restricting another type, say `α`, to its elements that satisfy some
+predicate, say `p : α → Prop`. Specifically, it is the type of pairs `⟨val, property⟩` where
+`val : α` and `property : p val`. It is denoted `Subtype p` and notation `{val : α // p val}` is
+available.
+
+A subtype has a natural coercion to the parent type, by coercing `⟨val, property⟩` to `val`. As
+such, subtypes can be thought of as bundled sets, the difference being that elements of a set are
+still of type `α` while elements of a subtype aren't.
+-/
+
 
 open Function
 
 namespace Subtype
-variable {α : Sort _} {β : Sort _} {γ : Sort _} {p : α → Prop} {q : α → Prop}
 
-def simps.coe (x: Subtype p) : α := x
+#align subtype Subtype
```

Comment:
This is unnecessary, I think. As far as I can see in the mathlib3port output, it's already happening correctly?